pypose/optim/controller_parameters_tuner.py

```diff
@@ -0,0 +1,101 @@
+import torch
+from torch import nn
+from torch.autograd.functional import jacobian
+
+
+class ControllerParametersTuner(nn.Module):
+    r"""
+    This class is the general implementation of the controller parameters tuner based on
+    the DiffTune paper.
+
+    Given a dynamic system, a corresponding controller and reference states, assuming all
+    these components are differentiable, it is possbile to find the most suitable
+    parameters by constructing the loss function $\mathbf{L}$ and computing the gradient of
+    $\mathbf{L}$ with respect to controller parameters.
+
+    Note:
+        For the parameter initial_state and parameters, please use row/column vector to
+        represent these variables at the same time.
+
+    .. math::
+        \begin{align*}
```

Comment:
Make the documentation more clear, with all the core algorithms and equations, then add the important citation. Refer to LQR and all other docs.